content/BatchRequestContentCollection.md

```diff
@@ -0,0 +1,23 @@
+# BatchRequestContentCollection
+
+## Objectives
+
+Provide a class that does not have limitation of that 20 requests by creating new `BatchRequestContent` objects to hold new requests to simplify the batching for advanced scenarios. 
+The `BatchRequestContentCollection` is essentially a collection wrapper for `BatchRequestContent` and is not expected to be serializable as the individual `BatchRequestContent` instance are sent out independently and will receive independent responses.
+
+### Requirements
+
+Similar to [BatchRequestContent](BatchRequestContent.md), it must support the below functionality while holding a collection of BatchRequestContent instances.
+
+- It MUST be possible to add native platform HTTP request objects to a BatchRequestContent object. The request URL MUST be a relative URL. Adding Graph specific Request objects MAY be supported.
+- Added requests MUST be assigned unique identifiers. Those identifiers MAY be user provided or autogenerated. The uniqueness SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in AND across the `BatchRequestContent` instances.
+- It MUST be possible to create dependencies between request objects. However, these dependencies SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in and not across `BatchRequestContent` instances.
+
+> Given the similar API experience between `BatchRequestContentCollection` and `BatchRequestContent`, `BatchRequestContent` SHOULD be used internally and not exposed publicly to avoid confusion for the end user. Adding new requests should automatically generate new `BatchRequestContent` instances to the `BatchRequestContentCollection`.
```

Comment:
Should we? Are there scenarios where developers would want to roll out their own version of the collection based on their own "rules"? For instance, handling optimal throttling on the Exchange workloads with a maximum of 4 requests / batch. If they want to do it, they would need to use the `BatchRequestContent`. I see this as an added-value scenario but it doesn't mean that we have to deprecate the other approach.

Reply:
Given @sebastienlevert thoughts, 
It probably does, making this internal may block customized scenarios that may come up due to workload quirks and user needs. I've updated the statement to reflect that documentation should encourage the use of the `BatchRequestContentCollection` to avoid confusion for the end user. 
Furthermore, `BatchRequestContentCollection` should allow the configuring of the maximum requests per `BatchRequestContent` as well